### DISCUSSION TITLE
Fn Key issue

### DISCUSSION THREAD

**Discussion by Linux-fuer-Einsteiger** [SpiralLinux/SpiralLinux-project#92] [Q&A]
Hello,

i have SpiralLinux on a Fujitsu Liifebook and got problems with the Fn keys. 
Id like to have to normal keyboard  but for example : when i press k there will be a 2 on my Desktop. I am from germany  and i already try to put in my german (qwertz keyboard) but it will not work. Do you have a solution for me? Thx forward

**Comment by geckolinux** [comment]
Hi there, that sounds like the NumLock function somehow got activated. Also it's possible that you selected the wrong keyboard layout in the Calamares installer.

**Reply by janxkoci** [reply]
I have the same issue every time I boot into SpiralLinux Budgie (bookworm version) - is there a config where I can disable NumLock after login? I tried to look in dconf editor and seen two settings that seemed relevant (for Cinnamon and GNOME desktops) but flipping the switch on either of them (or both) had no effect. (I can later look up what settings exactly they were, when I have the machine at hand.)

BTW I have an Acer laptop without dedicated numpad - instead it has a numpad as part of the regular keyboard, with blue symbols at some keys, which gets activated with Fn+NumLock. I think it's the same situation as OP has.

**Reply by geckolinux** [reply]
Hi there, there might be a BIOS setting on your laptop for NumLock, because that doesn't happen on any of the many laptops I have tested with SpiralLinux. But at any rate, you can also control it with software, install this package and set it to run in the startup programs:
https://manpages.debian.org/trixie/numlockx/numlockx.1.en.html

**Reply by janxkoci** [reply]
This laptop already had Spiral Linux MATE and XFCE, as well as several other distros in the past (elementaryOS, Ubuntu MATE, Xubuntu), and none of them (IIRC) were activating NumLock on login. So I don't think it's in the bios.

I know from the past that some beginner-friendly distros tend to activate NumLock, but I've only seen this with real dedicated numpad (I used to have external keyboard years ago). Now with Spiral Linux Budgie it's the first time I see this on a laptop keyboard with Fn-activated NumLock.

**Reply by marwenius** [reply]
[Based on this post](https://discourse.ubuntubudgie.org/t/numlock-is-forced-on-at-session-login-in-24-04/7311), there are two options that can be set:
```
gsettings set org.gnome.desktop.peripherals.keyboard numlock-state "false"
gsettings set org.gnome.desktop.peripherals.keyboard remember-numlock-state "true"
```
Additionally, you can install the [numlockx package](https://packages.debian.org/search?keywords=numlockx&exact=1) and set the value `NUMLOCK=off` in the configuration file `/etc/default/numlockx`.

**Reply by janxkoci** [reply]
Here is what I see in dconf editor:

<img width="828" height="599" alt="Snapshot_2025-11-26_21-38-55" src="https://github.com/user-attachments/assets/7c975d87-d62a-46f2-8ef6-ad320fb0cdfc" />

This is showing the state as I first opened it, but I tried to flip all the switches to no avail.

I'll try the numlockx package, bookworm seems to have it too..

PS: numlockx was already installed, and the config was set to `NUMLOCK=auto` - I've changed it to `off` and I'm gonna reboot, wish me luck... :crossed_fingers:

**Reply by janxkoci** [reply]
Hooray - it worked! :grin: 

> set the value `NUMLOCK=off` in the configuration file `/etc/default/numlockx`.

This solved the problem for me, thanks!

**Reply by geckolinux** [reply]
Good, thanks for confirming. It must be a Budgie thing that it is setting somewhere, I'm not surprised that the dconf switches didn't help, Budgie is a bit weird with its semi-dependence on Gnome while also having some independent settings of its own.

**Comment by Orcrist9691** [comment]
I also run Spiral Gnome on a Fujitsu Lifebook. The problem you have discribed is 99% the NumLock as already suggested.
